### ISSUE TITLE
Feature Request: Should ngMocks set signals mocks to singal(null) rather than null?

### ISSUE THREAD

**Issue by cford256** [help-me-mom/ng-mocks#6918]
Given I use a signal in an html template like below

```
@Component({
  selector: 'app-component',
  template: '<h1>{{ count() }}</h1>',
})
export class AppComponent{
   count = signal(3);
}
```
And I want to have a mock version of it when testing a different component. 

```
beforeAll(() => MockBuilder(OtherAppComponent, ItsModule) );

beforeEach(() => {
  const fixture = MockRender(OtherAppComponent);
});
```

Then I get an error along the lines of ctx.count is not a function. 
I believe this is because count is being set to null and then the template is trying to call it as a function. 
I am able to use Mock Instance to create a custom mock of the signal to avoid this. 
Was not sure if the way it currently works is the desired behavior. 

### Proposed solution
Instead of setting a signal mock to null, set it to signal(null).

**Comment by satanTime** [comment]
Hi @cford256,

that's a good FR. I'll take a look how it's possible to detect signals to mock them.

At this moment, I would recommend something like [default observables](https://ng-mocks.sudo.eu/extra/mock-observables#permanent-fix):

```
ngMocks.defaultMock(TodoService, () => ({
  count : signal(3),
}));
```

**Comment by cford256** [comment]
Good Idea, I think that I could simplify my code a bit with the help of that function.

**Comment by ninoeilQc** [comment]
Hi everyone, is there any update on this issue?

I would like to use MockService to mock my facade service (abstracting http layers and other services) using signals (state) and toSignal (converting http request observable) and use it with testing-library.

My structure is Component => Facade Service => Many others services like api services

Right now, I need to hard mock my service to prevent error like ctx.someSignal is not a function.

Thank you all

**Comment by satanTime** [comment]
Duplicate of #2701

The signal requirements are now included in #2701 alongside configurable Observable defaults and controllable reactive mocks. That issue will track ordinary `signal()`, `computed()`, and `toSignal()` fields, configurable initial values, and the shared limitation of discovering reactive members without executing the real implementation.

Closing this issue to consolidate the feature request; ordinary signal fields still require explicit mock customization. Angular signal inputs (`input()` / `input.required()`) are a separate supported case.

### Workaround with MockInstance today

Provide an actual signal for the mocked field before the dependency is instantiated or the component is rendered. For example, assuming `TargetComponent` reads `facade.count()`:

```ts
import { signal } from '@angular/core';
import { MockBuilder, MockInstance, MockRender } from 'ng-mocks';

describe('TargetComponent', () => {
  MockInstance.scope();

  beforeEach(() => MockBuilder(TargetComponent, TargetModule));

  beforeEach(() => {
    MockInstance(FacadeService, 'count', signal(0));
  });

  it('renders the configured signal value', () => {
    const fixture = MockRender(TargetComponent);

    expect(fixture.nativeElement.textContent).toContain('0');
  });
});
```

`MockInstance.scope()` restores the mock configuration between tests, and the `beforeEach` creates a fresh signal for each test. Choose an initial value appropriate for the consumer, and customize any other signal fields it reads in the same way. A field originally created with `computed()` or `toSignal()` can also be supplied with a signal containing the value needed by the test.

See the [MockInstance documentation](https://ng-mocks.sudo.eu/api/MockInstance/).
